frontend/common/src/components/Pagination/usePagination.ts

```diff
@@ -0,0 +1,118 @@
+import { useMemo, useState } from "react";
+
+export const DOTS = "...";
+
+const range = (start: number, end: number) => {
+  const length = end - start + 1;
+  /*
+  	Create an array of certain length and set the elements within it from
+    start value to end value.
+  */
+  return Array.from({ length }, (_, idx) => idx + start);
+};
+
+/**
+ * A pagination hook that returns the range of numbers to be displayed in out Pagination component as an array.
+ * @see https://www.freecodecamp.org/news/build-a-custom-pagination-component-in-react/
+ * @param totalCount
+ * @param pageSize
+ * @param siblingCount
+ * @param currentPage
+ * @returns (number | string)[]
+ */
+export const usePagination = ({
+  totalCount,
+  pageSize,
+  siblingCount = 1,
+  currentPage,
+}: {
+  totalCount: number;
+  pageSize: number;
+  siblingCount: number;
+  currentPage: number;
+}): (number | string)[] => {
+  const paginationRange = useMemo(() => {
```

Comment:
eslint would like a return statement at the end of this funciton.  Is there a `!shouldShowLeftDots && !shouldShowRightDots` case?  Maybe a throw error or something in case none of the expected conditions are hit?

Reply:
I added some error handling for negative values, and if none of the use cases get hit. Let me know what you think.